src/scvi/external/sysvi/_model.py

```diff
@@ -0,0 +1,290 @@
+                )
+            assert pseudoinputs_data_indices.shape[0] == n_prior_components
+            assert pseudoinputs_data_indices.ndim == 1
+            pseudoinput_data = next(
+                iter(
+                    self._make_data_loader(
+                        adata=adata,
+                        indices=pseudoinputs_data_indices,
+                        batch_size=n_prior_components,
+                        shuffle=False,
+                    )
+                )
+            )
+        else:
+            pseudoinput_data = None
+
+        n_cats_per_cov = (
+            self.adata_manager.get_state_registry(REGISTRY_KEYS.CAT_COVS_KEY).n_cats_per_key
+            if REGISTRY_KEYS.CAT_COVS_KEY in self.adata_manager.data_registry
+            else None
+        )
+
+        self.module = SysVAE(
+            n_input=self.summary_stats.n_vars,
+            n_batch=self.summary_stats.n_batch,
+            n_continuous_cov=self.summary_stats.get("n_extra_continuous_covs", 0),
+            n_cats_per_cov=n_cats_per_cov,
+            prior=prior,
+            n_prior_components=n_prior_components,
+            pseudoinput_data=pseudoinput_data,
+            **model_kwargs,
+        )
+
+        self._model_summary_string = (
+            "SysVI - cVAE model with optional VampPrior " + "and latent cycle-consistency loss."
+        )
+        self.init_params_ = self._get_init_params(locals())
+
+        logger.info("The model has been initialized")
+
+    def train(
+        self,
+        *args,
+        plan_kwargs: dict | None = None,
+        **kwargs,
+    ):
+        """Train the models.
+
+        Overwrites the ``train`` method of
+        class:`~scvi.model.base.UnsupervisedTrainingMixin`
+        to prevent the use of KL loss warmup (specified in ``plan_kwargs``).
+        This is disabled as our experiments showed poor integration in the
+        cycle model when using KL loss warmup.
+
+        Parameters
+        ----------
+        args
+            Training args.
+        plan_kwargs
+            Training plan kwargs.
+        kwargs
+            Training kwargs.
+        """
+        plan_kwargs = plan_kwargs or {}
+        kl_weight_defaults = {"n_epochs_kl_warmup": 0, "n_steps_kl_warmup": 0}
+        if any(v != plan_kwargs.get(k, v) for k, v in kl_weight_defaults.items()):
+            warnings.warn(
+                "The use of KL weight warmup is not recommended in SysVI. "
+                + "The n_epochs_kl_warmup and n_steps_kl_warmup "
+                + "will be reset to 0.",
+                stacklevel=settings.warnings_stacklevel,
+            )
+        # Overwrite plan kwargs with kl weight defaults
+        plan_kwargs = {**plan_kwargs, **kl_weight_defaults}
+
+        # Pass to parent
+        kwargs = kwargs or {}
+        kwargs["plan_kwargs"] = plan_kwargs
+        super().train(*args, **kwargs)
+
+    @torch.inference_mode()
+    def get_latent_representation(
```

Comment:
Does it need it's own get_latent_representation or can it inherit from VAEMIXIN class?